### PR TITLE
Make sure all contents in readme is included for secret management module

### PR DESCRIPTION
**Pull request by chenrujun**
Resolve https://github.com/Azure/azure-sdk-for-java/issues/25923

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @backwind1233 , @saragluna , please help to review this PR when you have time.

File: docs/src/main/asciidoc/secret-management.adoc [hunk 1]
```diff
@@ -56,23 +57,25 @@ spring:
           endpoint: ${AZURE_KEYVAULT_ENDPOINT}
 ----
 
+If your application is authenticated by other methods like Managed Identity or Azure CLI, properties like `tenant-id`, `client-id`, `client-secret` is not necessary. But if these properties are configured, then these properties have higher priority. Please refer to link:authentication.html[Authentication section] to get more information.
```
Comment (backwind1233) [inline docs/src/main/asciidoc/secret-management.adoc:60]: properties like `tenant-id`, `client-id`, `client-secret` is not necessary --> properties like `tenant-id`, `client-id`, `client-secret` are not necessary

File: docs/src/main/asciidoc/secret-management.adoc [hunk 2]
```diff
@@ -56,23 +57,25 @@ spring:
...
+
 ===== Java code
 
 [source,java]
 ----
 @SpringBootApplication
 public class KeyVaultSample implements CommandLineRunner {
```
Comment (backwind1233) [inline docs/src/main/asciidoc/secret-management.adoc:68]: Have you tested it?

File: docs/src/main/asciidoc/secret-management.adoc (hunk 1 above)
Comment (chenrujun) [reply docs/src/main/asciidoc/secret-management.adoc]: Updated.

File: docs/src/main/asciidoc/secret-management.adoc (hunk 2 above)
Comment (chenrujun) [reply docs/src/main/asciidoc/secret-management.adoc:68]: No, it's just code snippet, not project, no need to test, I think.

File: docs/src/main/asciidoc/secret-management.adoc [hunk 3]
```diff
@@ -116,26 +120,55 @@ spring:
...
+
+[source,properties]
+----
+spring.cloud.azure.keyvault.secret.property-sources[].case-sensitive=true
+----
+
+
 === Samples
 
 Please refer to link:https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_4.0[azure-spring-boot-samples] for more details.
```
Comment (backwind1233) [inline docs/src/main/asciidoc/secret-management.adoc:174]: I think it's better to link the keyvault samples here.

File: docs/src/main/asciidoc/secret-management.adoc [hunk 4]
```diff
@@ -116,26 +120,55 @@ spring:
 @SpringBootApplication
 public class SampleApplication implements CommandLineRunner {
 
-    @Value("${secret-name-in-key-vault-1}")
-    private String secretNameInKeyVault1;
-    @Value("${secret-name-in-key-vault-2}")
-    private String secretNameInKeyVault2;
-    @Value("${secret-name-in-key-vault-both}")
-    private String secretNameInKeyVaultBoth;
+    @Value("${sampleProperty1}")
```
Comment (backwind1233) [inline docs/src/main/asciidoc/secret-management.adoc:123]: Why do you change the names here, if you are willing to do this change, will you also update the names in samples？
image

File: docs/src/main/asciidoc/secret-management.adoc [hunk 5]
```diff
@@ -116,26 +120,55 @@ spring:
...
+=== Advanced Usage
+
+==== Special characters in property name
+Key Vault secret name only support characters in `[0-9a-zA-Z-]`. Refs: link:https://docs.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#vault-name-and-object-name[Vault-name and Object-name]. If your property name contains other characters, you can use these workarounds:
+
+===== Use `-` instead of `.` in secret name
+
+`.` is not supported in secret name. If your application have property name which contain `.`, like `spring.datasource.url`, just replace `.` to `-` when save secret in Azure Key Vault. For example: Save `spring-datasource-url` in Azure Key Vault. In your application, you can still use `spring.datasource.url` to retrieve property value.
+
+===== Use property placeholders
```
Comment (backwind1233) [inline docs/src/main/asciidoc/secret-management.adoc:152]: An other question is,
 I think `====` is for title, do you think we should use the same pattern(Style) for titles, refer [here](https://capitalizemytitle.com/#).
image

File: docs/src/main/asciidoc/secret-management.adoc (hunk 4 above)
Comment (chenrujun) [reply docs/src/main/asciidoc/secret-management.adoc:123]: PR created: https://github.com/Azure-Samples/azure-spring-boot-samples/pull/151

File: docs/src/main/asciidoc/secret-management.adoc (hunk 3 above)
Comment (chenrujun) [reply docs/src/main/asciidoc/secret-management.adoc]: Updated.

File: docs/src/main/asciidoc/secret-management.adoc (hunk 5 above)
Comment (chenrujun) [reply docs/src/main/asciidoc/secret-management.adoc]: Updated.

And I created an issue to update all titles: https://github.com/Azure/azure-sdk-for-java/issues/26458

**Comment by chenrujun** [comment]
Closing this PR, use #803 instead. #803 is targeting to 4.0.0-beta.3 version

**Pull request closed without merging**